esp-hal-common/devices/esp32.toml

```diff
@@ -53,4 +53,9 @@ peripherals = [
     "dac",
     "pdma",
     "radio",
+
+    # Wakeup SOC based on ESP-IDF:
+    "SOC_PM_SUPPORT_EXT_WAKEUP",
```

Comment:
Not sure if we have a style for these, the others are lower case + underscores, maybe it makes sense to do the same here?

Reply:
Yeah, I would very much like to avoid adding these non-peripheral name symbols at all. They're kind of a workaround as it is, and I would like to eventually get rid of them altogether if possible.

Reply:
also, maybe the "SOC" prefix is not really needed? Don't have a strong opinion on that t.b.h

Reply:
> Yeah, I would very much like to avoid adding these non-peripheral name symbols at all. They're kind of a workaround as it is, and I would like to eventually get rid of them altogether if possible.
> 
> Need to document this, I guess.

I think we should have those and add more - most probably not as `peripheral` but maybe `capabilities` ... naming is hard - I can think of a lot of places where these might make things easier in future and cleaner.

I mean things like here: https://github.com/esp-rs/esp-hal/blob/15ce2d1452e540b1b12b49594fedbcb96ad9382a/esp-hal-common/src/spi.rs#L2818-L2832 where we currently need to do conditional compilation based on the concrete target. Something like `spi_needs_update` (again a terrible name) would be easier to understand - there are many places like this

Reply:
I updated PR and let's wait for @jessebraham to agree on what to do with the symbols.